lib/packages.js

```diff
@@ -378,11 +378,19 @@ function _sortPackages(cache) {
   // sort packages, least dependent => most dependent (low-level to high-level)
   const unsorted = Object.keys(cache).map(name => cache[name]);
   const sorted = [];
+  const cyclicHashSet = {};
```

Comment:
Could use `const cyclicHashSet = new Set();` instead ... and check `cyclicHashSet.has(foo)` below.